### ISSUE TITLE
compliance: §6.6.2.3.1 Tagged PDF structure tree violations not detected — 17 false negatives

### ISSUE THREAD

**Issue by jasperdew** [jasperdew/xfa-native-rust#453]
## Summary
The compliance checker does not detect **§6.6.2.3.1** (structure element attribute objects) and **§6.9** (XMP metadata schemas) violations. 20 PDFs from the veraPDF tagged test suite pass our checker but are rejected by veraPDF.

## Impact
20 false negatives in 4K corpus (run-20260315-065448.905)

## Failing PDFs
```
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t01-fail-e.pdf  → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t01-fail-j.pdf  → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t01-fail-u.pdf  → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t02-fail-t.pdf  → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t04-fail-e.pdf  → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t08-fail-c.pdf  → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t13-fail-d/e/f.pdf → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t14-fail-b/d/q/x.pdf → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t16-fail-i/q.pdf → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-1-t17-fail-r/t.pdf → ["6.6.2.3.1"]
tagged/verapdf/veraPDF test suite 6-6-2-3-3-t01-fail-f.pdf  → ["6.6.2.3.3"]
tagged/verapdf/veraPDF test suite 6-3-1-t01-fail-e.pdf       → ["6.9"]
tagged/verapdf/veraPDF test suite 6-9-t04-fail-a.pdf         → ["6.9"]
```

## Spec
- **§6.6.2.3.1**: Structure element `/A` attribute objects must use owners from the standard set (`Layout`, `List`, `PrintField`, `Table`, `Artifact`, `XML-1.00`)
- **§6.6.2.3.3**: `/Revision` attribute must be a non-negative integer
- **§6.9**: Required XMP metadata schemas must be present and well-formed

## Fix location
`crates/pdf-compliance/src/check.rs` — §6.6 and §6.9 sections.
Parse `/StructTreeRoot`, traverse structure elements, validate `/A` owners against the standard list.
